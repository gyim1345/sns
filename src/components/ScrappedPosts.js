import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import Swal from 'sweetalert2';

import { css } from '@emotion/core';

import { scrappedPosts } from '../apis/scrap';
import PostImagesOnly from './PostImagesOnly';

function ScrappedPosts({ user }) {
  const [posts, setPosts] = useState([]);

  const ScrappedPosts = async () => {
    try {
      const response = await scrappedPosts(user);
      setPosts(response);
    } catch (e) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Internal Error'
      });
    }
  };

  useEffect(() => {
    ScrappedPosts();
  }, []);

  return (
    <div css={[tagPostImagesCss]}>
      {posts.map(post => (
        <Link to={`/posting/${post.id}`} key={post.id}>
          <PostImagesOnly imageUrl={post.imageUrl} />
        </Link>
      ))}
    </div>
  );
}

const tagPostImagesCss = css`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  flex-basis: 33%;
  justify-content: center;
`;

export default ScrappedPosts;
