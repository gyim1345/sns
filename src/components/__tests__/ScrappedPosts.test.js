import React from 'react';
import { mount } from 'enzyme';
import ScrappedPosts from '../ScrappedPosts';
import { currentUser } from './testjs';

describe('<ScrappedPosts />', () => {
  it('스냅샷 비교', () => {
    const wrapper = mount(<ScrappedPosts user={currentUser} />);
    expect(wrapper).toMatchSnapshot();
  });
});
