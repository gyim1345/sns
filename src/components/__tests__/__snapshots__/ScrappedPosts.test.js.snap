// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrappedPosts /> 스냅샷 비교 1`] = `
<ScrappedPosts
  user="gibong@gmail.com"
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="ScrappedPosts"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlNjcmFwcGVkUG9zdHMuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBd0M0QiIsImZpbGUiOiJTY3JhcHBlZFBvc3RzLmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0LCB7IHVzZVN0YXRlLCB1c2VFZmZlY3QgfSBmcm9tICdyZWFjdCc7XG5pbXBvcnQgeyBMaW5rIH0gZnJvbSAncmVhY3Qtcm91dGVyLWRvbSc7XG5pbXBvcnQgU3dhbCBmcm9tICdzd2VldGFsZXJ0Mic7XG5cbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuXG5pbXBvcnQgeyBzY3JhcHBlZFBvc3RzIH0gZnJvbSAnLi4vYXBpcy9zY3JhcCc7XG5pbXBvcnQgUG9zdEltYWdlc09ubHkgZnJvbSAnLi9Qb3N0SW1hZ2VzT25seSc7XG5cbmZ1bmN0aW9uIFNjcmFwcGVkUG9zdHMoeyB1c2VyIH0pIHtcbiAgY29uc3QgW3Bvc3RzLCBzZXRQb3N0c10gPSB1c2VTdGF0ZShbXSk7XG5cbiAgY29uc3QgU2NyYXBwZWRQb3N0cyA9IGFzeW5jICgpID0+IHtcbiAgICB0cnkge1xuICAgICAgY29uc3QgcmVzcG9uc2UgPSBhd2FpdCBzY3JhcHBlZFBvc3RzKHVzZXIpO1xuICAgICAgc2V0UG9zdHMocmVzcG9uc2UpO1xuICAgIH0gY2F0Y2ggKGUpIHtcbiAgICAgIFN3YWwuZmlyZSh7XG4gICAgICAgIGljb246ICdlcnJvcicsXG4gICAgICAgIHRpdGxlOiAnT29wcy4uLicsXG4gICAgICAgIHRleHQ6ICdJbnRlcm5hbCBFcnJvcidcbiAgICAgIH0pO1xuICAgIH1cbiAgfTtcblxuICB1c2VFZmZlY3QoKCkgPT4ge1xuICAgIFNjcmFwcGVkUG9zdHMoKTtcbiAgfSwgW10pO1xuXG4gIHJldHVybiAoXG4gICAgPGRpdiBjc3M9e1t0YWdQb3N0SW1hZ2VzQ3NzXX0+XG4gICAgICB7cG9zdHMubWFwKHBvc3QgPT4gKFxuICAgICAgICA8TGluayB0bz17YC9wb3N0aW5nLyR7cG9zdC5pZH1gfSBrZXk9e3Bvc3QuaWR9PlxuICAgICAgICAgIDxQb3N0SW1hZ2VzT25seSBpbWFnZVVybD17cG9zdC5pbWFnZVVybH0gLz5cbiAgICAgICAgPC9MaW5rPlxuICAgICAgKSl9XG4gICAgPC9kaXY+XG4gICk7XG59XG5cbmNvbnN0IHRhZ1Bvc3RJbWFnZXNDc3MgPSBjc3NgXG4gIGRpc3BsYXk6IGZsZXg7XG4gIGZsZXgtZGlyZWN0aW9uOiByb3c7XG4gIGZsZXgtd3JhcDogd3JhcDtcbiAgZmxleC1iYXNpczogMzMlO1xuICBqdXN0aWZ5LWNvbnRlbnQ6IGNlbnRlcjtcbmA7XG5cbmV4cG9ydCBkZWZhdWx0IFNjcmFwcGVkUG9zdHM7XG4iXX0= */",
        "name": "10obuon-tagPostImagesCss-ScrappedPosts",
        "next": undefined,
        "styles": "display:flex;flex-direction:row;flex-wrap:wrap;flex-basis:33%;justify-content:center;;label:tagPostImagesCss;;;label:ScrappedPosts;/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlNjcmFwcGVkUG9zdHMuanMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBOEJTIiwiZmlsZSI6IlNjcmFwcGVkUG9zdHMuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QsIHsgdXNlU3RhdGUsIHVzZUVmZmVjdCB9IGZyb20gJ3JlYWN0JztcbmltcG9ydCB7IExpbmsgfSBmcm9tICdyZWFjdC1yb3V0ZXItZG9tJztcbmltcG9ydCBTd2FsIGZyb20gJ3N3ZWV0YWxlcnQyJztcblxuaW1wb3J0IHsgY3NzIH0gZnJvbSAnQGVtb3Rpb24vY29yZSc7XG5cbmltcG9ydCB7IHNjcmFwcGVkUG9zdHMgfSBmcm9tICcuLi9hcGlzL3NjcmFwJztcbmltcG9ydCBQb3N0SW1hZ2VzT25seSBmcm9tICcuL1Bvc3RJbWFnZXNPbmx5JztcblxuZnVuY3Rpb24gU2NyYXBwZWRQb3N0cyh7IHVzZXIgfSkge1xuICBjb25zdCBbcG9zdHMsIHNldFBvc3RzXSA9IHVzZVN0YXRlKFtdKTtcblxuICBjb25zdCBTY3JhcHBlZFBvc3RzID0gYXN5bmMgKCkgPT4ge1xuICAgIHRyeSB7XG4gICAgICBjb25zdCByZXNwb25zZSA9IGF3YWl0IHNjcmFwcGVkUG9zdHModXNlcik7XG4gICAgICBzZXRQb3N0cyhyZXNwb25zZSk7XG4gICAgfSBjYXRjaCAoZSkge1xuICAgICAgU3dhbC5maXJlKHtcbiAgICAgICAgaWNvbjogJ2Vycm9yJyxcbiAgICAgICAgdGl0bGU6ICdPb3BzLi4uJyxcbiAgICAgICAgdGV4dDogJ0ludGVybmFsIEVycm9yJ1xuICAgICAgfSk7XG4gICAgfVxuICB9O1xuXG4gIHVzZUVmZmVjdCgoKSA9PiB7XG4gICAgU2NyYXBwZWRQb3N0cygpO1xuICB9LCBbXSk7XG5cbiAgcmV0dXJuIChcbiAgICA8ZGl2IGNzcz17W3RhZ1Bvc3RJbWFnZXNDc3NdfT5cbiAgICAgIHtwb3N0cy5tYXAocG9zdCA9PiAoXG4gICAgICAgIDxMaW5rIHRvPXtgL3Bvc3RpbmcvJHtwb3N0LmlkfWB9IGtleT17cG9zdC5pZH0+XG4gICAgICAgICAgPFBvc3RJbWFnZXNPbmx5IGltYWdlVXJsPXtwb3N0LmltYWdlVXJsfSAvPlxuICAgICAgICA8L0xpbms+XG4gICAgICApKX1cbiAgICA8L2Rpdj5cbiAgKTtcbn1cblxuY29uc3QgdGFnUG9zdEltYWdlc0NzcyA9IGNzc2BcbiAgZGlzcGxheTogZmxleDtcbiAgZmxleC1kaXJlY3Rpb246IHJvdztcbiAgZmxleC13cmFwOiB3cmFwO1xuICBmbGV4LWJhc2lzOiAzMyU7XG4gIGp1c3RpZnktY29udGVudDogY2VudGVyO1xuYDtcblxuZXhwb3J0IGRlZmF1bHQgU2NyYXBwZWRQb3N0cztcbiJdfQ== */",
        "toString": [Function],
      }
    }
  >
    <div
      className="css-10obuon-tagPostImagesCss-ScrappedPosts"
    />
  </EmotionCssPropInternal>
</ScrappedPosts>
`;
